### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by NicolasLM**
* Document how to use IDLE
* Write a note about thread safety
* Split installation into its own page
* Simplify a bit the README
* Mention that `backports.ssl` is just a compatibility layer

### CODE REVIEW

File: doc/src/advanced.rst [hunk 1]
```diff
@@ -30,9 +30,23 @@ connections when they are not needed anymore::
       c.login("bar@foo.org", "passwd")
       c.select_folder("INBOX")
 
-Watching a mailbox asynchronously using idle
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-TODO
+Watching a mailbox using idle
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The IDLE extension allows an IMAP server to notify a client when something
+changes in a mailbox. It can be used as an alternative to polling to receive
```
Comment (mlorant) [inline doc/src/advanced.rst]: Maybe emphasise the "selected" mailbox notion (e.g. a sent message appended in INBOX/Sent won't end the IDLE if you're watching the INBOX)

File: README.rst [hunk 2]
```diff
@@ -64,18 +64,17 @@ point in the future.
...
-    easy_install IMAPClient
-
 More installation methods are described in the documentation.
 
 Documentation
 -------------
-IMAPClient's manual is available at http://imapclient.readthedocs.io/. Release notes can be found at http://imapclient.readthedocs.io/#release-history.
+IMAPClient's manual is available at http://imapclient.readthedocs.io/.
+Release notes can be found at
+http://imapclient.readthedocs.io/#release-history.
```
Comment (mlorant) [inline README.rst]: This makes me think the releases notes is not up to date for the 2.0. Maybe in another PR.

File: doc/src/advanced.rst [hunk 3]
```diff
@@ -136,3 +150,8 @@ oauth2_refresh_token string  OAUTH2 token for refreshing the secret.
 
 Acceptable boolean values are "1", "yes", "true", and "on", for true;
 and "0", "no", "false", and "off", for false.
+
+Thread Safety
```
Comment (mlorant) [inline doc/src/advanced.rst]: While short, great reminder :+1:

File: doc/src/advanced.rst (hunk 1 above)
Comment (NicolasLM) [reply doc/src/advanced.rst]: Done

**Pull request merged**